sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/GenericArrayData.scala

```diff
@@ -37,6 +37,11 @@ class GenericArrayData(val array: Array[Any]) extends ArrayData {
   def this(primitiveArray: Array[Byte]) = this(primitiveArray.toSeq)
   def this(primitiveArray: Array[Boolean]) = this(primitiveArray.toSeq)
 
+  def this(seqOrArray: Any) = this(seqOrArray match {
```

Comment:
We don't. For `RowEncoder`, we only have the schema when we build it, no type info. Actually there are 3 special cases overall, one is array type, we support both seq and array. One is decimal type, we support both java/scala `BigDecimal` and `Decimal`. One is struct type, we support both `Row` and `Product`. They all use runtime reflection